examples/kafkaconsumer/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/wvanbergen/kafka/kafkaconsumer"
+)
+
+const (
+	DefaultKafkaTopics   = "test_topic"
+	DefaultConsumerGroup = "consumer_example.go"
+)
+
+var (
+	consumerGroup  = flag.String("group", DefaultConsumerGroup, "The name of the consumer group, used for coordination and load balancing")
+	kafkaTopicsCSV = flag.String("topics", DefaultKafkaTopics, "The comma-separated list of topics to consume")
+	zookeeper      = flag.String("zookeeper", "", "A comma-separated Zookeeper connection string (e.g. `zookeeper1.local:2181,zookeeper2.local:2181,zookeeper3.local:2181`)")
+)
+
+func init() {
+	sarama.Logger = log.New(os.Stdout, "[sarama]        ", log.LstdFlags|log.Lshortfile)
+	kafkaconsumer.Logger = log.New(os.Stdout, "[kafkaconsumer] ", log.LstdFlags|log.Lshortfile)
+}
+
+func main() {
+	flag.Parse()
+
+	if *zookeeper == "" {
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	config := kafkaconsumer.NewConfig()
+	config.Offsets.Initial = sarama.OffsetNewest
+
+	subscription := kafkaconsumer.TopicSubscription(strings.Split(*kafkaTopicsCSV, ",")...)
+	consumer, err := kafkaconsumer.Join(*consumerGroup, subscription, *zookeeper, config)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		<-c
+		if err := consumer.Close(); err != nil {
+			log.Println("Error closing the consumer", err)
+		}
+	}()
+
+	go func() {
+		for err := range consumer.Errors() {
+			log.Println(err)
+		}
+	}()
+
+	eventCount := 0
+	offsets := make(map[string]map[int32]int64)
+
+	for message := range consumer.Messages() {
+		if offsets[message.Topic] == nil {
+			offsets[message.Topic] = make(map[int32]int64)
+		}
+
+		eventCount += 1
+		if offsets[message.Topic][message.Partition] != 0 && offsets[message.Topic][message.Partition] != message.Offset-1 {
```

Comment:
This was copy pasted from the previous example app and is shitty. This also fails when you start consuming a partition, then another instance takes it over, and later you get the partition assigned back to you.

Will remove.